src/models/Link.php

```diff
@@ -583,7 +583,24 @@ public function getLinkURL()
                         $LinkURL = false;
                     }
                     if ($component->hasMethod('Link')) {
-                        $LinkURL = $component->Link() . $this->Anchor;
+                        $LinkURL = $component->Link();
+
+                        // add QueryString
+                        if(!empty($this->QueryString)) {
+                            if(strpos($LinkURL, '?') === false && strpos($this->QueryString, '?') === false) {
```

Comment:
strpos should be `str_contains`

The check on the QueryString seems pointless, as this is stripped out already earlier.